src/helm/ralph/templates/secrets.yaml

```
@@ -0,0 +1,59 @@
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Values.ralph.secret_name }}
+  namespace: {{ .Values.namespace }}
+  labels:
+    {{- include "ralph.labels" . | nindent 4 }}
+  annotations:
+    checksum/config: {{ (tpl (.Files.Glob "vault.yaml").AsSecrets . ) | sha256sum }}
+type: Opaque
+data:
+{{- $v := $.Files.Get "vault.yaml" | fromYaml }}
+{{- range $key, $val := $v }}
+    {{ if contains "RALPH_" $key }}
+        {{ $key | nindent 2 }}: {{ $val | b64enc }}
+    {{- end }}
+{{- end }}
+
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Values.ralph_lrs.auth_secret_name }}
+  namespace: {{ .Values.namespace }}
+  labels:
+    {{- include "ralph.labels" . | nindent 4 }}
+  annotations:
+    checksum/config: {{ (tpl (.Files.Glob "vault.yaml").AsSecrets . ) | sha256sum }}
+type: Opaque
+data:
+{{- $v := $.Files.Get "vault.yaml" | fromYaml }}
+{{- range $key, $val := $v }}
+    {{ if eq "LRS_AUTH" $key }}
+        {{ "auth.json:" | nindent 2 }}: {{ $val | toJson | b64enc }}
+    {{- end }}
+{{- end }}
```

Can we do something like the following?

```suggestion
{{ "auth.json:" | nindent 2 }}: {{ get $v "LRS_AUTH" | toJson | b64enc }}
```

---

I'd love to, but this is required because of the logic that you added in the tray, the vault file with secret value, I tried to implement the same logic here. Yes, this is a bit over designed in Helm and we may fix it if we use a normal var file. We can even prevent to commit it and add it as requirement to deploy, and in `values.yaml` we add only empty values.

---

fixed